OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -98,23 +101,22 @@ public void Flash()
 
 		public IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
-			if (needRenderables)
+			if (needRenderables && (renderables == NoRenderables || DamageStateCached != DamageState || OwnerCached != Owner))
 			{
-				needRenderables = false;
-				if (!actor.Destroyed)
-				{
-					IsRendering = true;
-					renderables = actor.Render(wr).ToArray();
-					IsRendering = false;
-				}
+				IsRendering = true;
```

Comment:
Please only use the ActorPreview if `actor.Destroyed` is true.  Using it for a living actor breaks the feature where the actor is frozen with its last visible frame (including muzzle flashes, turret position, etc).